code/modules/mapping/mapping_helpers.dm

```diff
@@ -245,6 +245,131 @@
 	else
 		airlock.autoname = TRUE
 
+//apc helpers
+/obj/effect/mapping_helpers/apc
+	desc = "You shouldn't see this. Report it please."
+	layer = DOOR_HELPER_LAYER
+	late = TRUE
+
+/obj/effect/mapping_helpers/apc/Initialize(mapload)
+	. = ..()
+	if(!mapload)
+		log_mapping("[src] spawned outside of mapload!")
+		return
+
+	var/obj/machinery/power/apc/apc = locate(/obj/machinery/power/apc) in loc
+	if(!apc)
+		log_mapping("[src] failed to find an apc at [AREACOORD(src)]")
+	else
+		payload(apc)
+
+/obj/effect/mapping_helpers/apc/LateInitialize()
+	. = ..()
+	var/obj/machinery/power/apc/apc = locate(/obj/machinery/power/apc) in loc
+
+	if(!apc)
+		qdel(src)
+		return
+	if(apc.cuttedAIwire)
+		apc.wires.cut(WIRE_AI)
+	if(apc.cell_5k)
+		apc.install_cell_5k()
+	if(apc.cell_10k)
+		apc.install_cell_10k()
+	if(apc.unlocked)
+		apc.unlock()
+	if(apc.syndicate_access)
+		apc.give_syndicate_access()
+	if(apc.away_general_access)
+		apc.give_away_general_access()
+	if(apc.no_charge)
+		apc.set_no_charge()
+	if(apc.full_charge)
+		apc.set_full_charge()
+	if(apc.cell_5k & apc.cell_10k)
+		CRASH("Non-combinable apc helpers!")
+	if(apc.syndicate_access & apc.away_general_access)
+		CRASH("Non-combinable apc helpers!")
+	if(apc.no_charge & apc.full_charge)
+		CRASH("Non-combinable apc helpers!")
+	apc.update_appearance()
+	qdel(src)
+
+/obj/effect/mapping_helpers/apc/proc/payload(obj/machinery/power/apc/apc)
+	return
+
+/obj/effect/mapping_helpers/apc/cuttedAIwire
+	name = "apc ai wire disabled helper"
+	icon_state = "apc_cutted_AIwire_helper"
+
+/obj/effect/mapping_helpers/apc/cuttedAIwire/payload(obj/machinery/power/apc/apc)
+	if(apc.cuttedAIwire)
+		log_mapping("[src] at [AREACOORD(src)] tried to mend the ai wire on the [apc] but it's already cut!")
```

Comment:
```suggestion
		var/area/apc_area = get_area(apc)
		log_mapping("[src] at [AREACOORD(src)] [(apc_area.type)] tried to mend the AI wire on the [apc] but it's already cut!")
```

Adding the typepath of the area is important because in case it's in a ruin or something it's much easier to find and locate this bug.